astropy/wcs/wcsapi/fitswcs.py

```diff
@@ -418,10 +418,17 @@ def _get_components_and_classes(self):
             if np.isnan(self.wcs.obsgeo[0]):
                 observer = None
             else:
-
                 earth_location = EarthLocation(*self.wcs.obsgeo[:3], unit=u.m)
-                obstime = Time(self.wcs.mjdobs, format='mjd', scale='utc',
-                               location=earth_location)
+
+                # Get the time scale from TIMESYS or fall back to 'utc'
+                tscale = self.wcs.timesys or 'utc'
+
+                if np.isnan(self.wcs.mjdavg):
+                    obstime = Time(self.wcs.mjdobs, format='mjd', scale=tscale,
+                                   location=earth_location)
+                else:
+                    obstime = Time(self.wcs.mjdavg, format='mjd', scale=tscale,
```

Comment:
Why does codecov think this line hasn't been covered? The test clearly should be hitting this, is there some subtle weirdness going on here? or did codecov :exploding_head:

Reply:
It only counts coverage on certain jobs. So is this the test covering this line supposed to be skipped under certain conditions (e.g., remote data)?

Reply:
If not, then the test is not doing what you think it is doing. I think we should make sure before merge.

Reply:
I added checks in the test that show that when a certain kw is `nan` the fall back is used. Regardless of what codecov says that line should be tested if i didn't miss something.

Reply:
To be sure, you can run coverage locally and double check. I don't see why codecov would miss this if this test is run regardless of pytest settings.

Reply:
Looks like we have two jobs that measure coverage now:

https://github.com/astropy/astropy/blob/1c5ce68e4b9569fc69d3e363a5e9f27741e95937/.github/workflows/ci_workflows.yml#L59-L62

https://github.com/astropy/astropy/blob/1c5ce68e4b9569fc69d3e363a5e9f27741e95937/.github/workflows/ci_workflows.yml#L74-L78